File: Data Structures & Algorithms/binary-search/submission-0.js
class Solution {
    search(nums, target) {
        let mid = 0;
        let start = 0;
        let end = nums.length - 1;
        for (let i = 0; i< nums.length;i++){
            mid = Math.floor(end+start/2);
            if (nums[mid] == target) {
                return mid;
            }else if (nums[mid] < target) {
                start = mid+1;
            }else{
                end = mid-1;
            }
        }    
        return -1;        
    }
}

File: Data Structures & Algorithms/combination-target-sum/submission-0.py
class Solution:
    def combinationSum(self, nums: List[int], t: int) -> List[List[int]]:
        result = []
        path = []

        def dfs(start, target):
            if target == 0:
                result.append(path[:])
                return
            if target < 0:
                return

            for i in range(start, len(nums)):
                curr = nums[i]
                path.append(curr)
                dfs(i,  target - curr)
                path.pop()
        
        dfs(0, t)
        return result